File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py

def solution(numbers, target):
    answer=0
    N = len(numbers)
    def dfs(depth,SUM):
        nonlocal answer
        if depth == N:
            if SUM==target:
                answer+=1
            return
        
        dfs(depth+1,SUM+numbers[depth])
        dfs(depth+1,SUM-numbers[depth])
        
            
    dfs(0,0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
class Solution {
    public int solution(int[] numbers, int target) {
        return count(numbers,target,0,0);
    }
    
    private int count(int[] numbers, int target, int idx, int sum){
        if (idx>=numbers.length){
            return sum==target ? 1 : 0;
        }
        int plus = count(numbers,target,idx+1,sum+numbers[idx] );
        int minus = count(numbers,target,idx+1,sum-numbers[idx]);
        return plus+minus;
    }
}